### PR TITLE
pkg/flare: add missing APM variables to envvars

### PR DESCRIPTION
**Pull request by hannahkm**
### What does this PR do?

This PR adds all APM environmental variables currently being used by the agent to the flare.

### Motivation

The lack of certain APM envvars lead to inconsistent documentation between internal docs and code. Adding these in can improve debugging with the amount of information contained by a flare.

### Describe how to test/QA your changes

Set those new env vars and collect a flare, make sure they're included in it.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by ogaca-dd — APPROVED** [review]
LGTM

File: pkg/flare/envvars.go [hunk 1]
```diff
@@ -69,6 +69,31 @@ var allowedEnvvarNames = []string{
 	"DD_APM_ADDITIONAL_ENDPOINTS",
 	"DD_APM_PROFILING_DD_URL",
 	"DD_APM_PROFILING_ADDITIONAL_ENDPOINTS",
+	"DD_APM_WINDOWS_PIPE_BUFFER_SIZE",
+	"DD_APM_WINDOWS_PIPE_SECURITY_DESCRIPTOR",
```
Comment (katiehockman) [inline pkg/flare/envvars.go:73]: Let's drop this one. I doubt anyone will actually need it in a flare.

File: pkg/flare/envvars.go [hunk 2]
```diff
@@ -69,6 +69,31 @@ var allowedEnvvarNames = []string{
...
+	"DD_APM_FILTER_TAGS_REQUIRE",
+	"DD_APM_FILTER_TAGS_REJECT",
+	"DD_APM_INTERNAL_PROFILING_ENABLED",
+	"DD_APM_DEBUGGER_DD_URL",
+	"DD_APM_DEBUGGER_ADDITIONAL_ENDPOINTS",
+	"DD_APM_SYMDB_DD_URL",
+	"DD_APM_SYMDB_ADDITIONAL_ENDPOINTS",
+	"DD_APM_TELEMETRY_ENABLED",
+	"DD_APM_TELEMETRY_DD_URL",
+	"DD_APM_TELEMETRY_ADDITIONAL_ENDPOINTS",
```
Comment (katiehockman) [inline pkg/flare/envvars.go:93]: I don't think these will be needed in a flare either, since customers shouldn't have to worry about telemetry at all.

**Review by ahmed-mez — COMMENTED** [review]
I'd drop all the `_ADDITIONAL_ENDPOINTS` ones as they might contain secrets.

**Comment by hannahkm** [comment]
> I'd drop all the `_ADDITIONAL_ENDPOINTS` ones as they might contain secrets.

Just wanted to confirm; some of the _ADDITIONAL_ENDPOINTS vars were included in the envvars file before this PR. I assume these should be kept? Thanks :)

**Comment by ahmed-mez** [comment]
> Just wanted to confirm; some of the _ADDITIONAL_ENDPOINTS vars were included in the envvars file before this PR. I assume these should be kept? Thanks :)

Let's be safe and remove all of them! You can see in the [code documentation](https://github.com/DataDog/datadog-agent/blob/456b3ad3a4436bafec3dfb3634fce82426554da2/pkg/trace/config/config.go#L261-L262) that those fields are supposed to contain API keys.

**Review by ahmed-mez — APPROVED** [review]
Looks good, thanks! Can you just mention in the PR description why the skip-qa label was applied + why there is no QA section?

**Comment by hannahkm** [comment]
> Looks good, thanks! Can you just mention in the PR description why the skip-qa label was applied + why there is no QA section?

Yes, SG! 👍🏼

**Review by katiehockman — APPROVED** [review]

**Pull request merged**